src/shogun/classifier/Perceptron.cpp

```diff
@@ -60,7 +60,7 @@ void CPerceptron::iteration()
 	bool converged = true;
 	SGVector<float64_t> w = get_w();
 
-	auto labels = binary_labels(m_labels)->get_int_labels();
+	auto labels = m_labels->as<CBinaryLabels>()->get_int_labels();
```

Comment:
reverting the old mapping approach which was invasive. Now algorithms simply assume they are given the right labels (as the case before any mapping was done)

Reply:
looks good!